### PR TITLE
saves cleanup

### PR DESCRIPTION
**Pull request by stohrendorf**

### CODE REVIEW

**Comment by serishema** [comment]
When I tested `deleteSavesPerLevelExceptNewest` by making consecutive saves in caves and then activiating the option, it did not delete any saves. 
I expected the eariler saves for that level to be deleted.

**Comment by stohrendorf** [comment]
Should be fixed now.

**Comment by serishema** [comment]
> Should be fixed now.

No luck. d902720 deletes a large number of saves, including the one highlighted when activating the deleteSavesExceptNewest option, while leaving saves that should have been deleted behind.

File: src/menu/savegamecleanup.cpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  for(const auto& [slot, savegameInfo] : savegameInfos)
+  {
+    const auto& newestLevelTs = newestPerLevel.at(savegameInfo.meta.filename);
+    if(savegameInfo.saveTime < newestLevelTs.timeStamp)
+    {
+      std::error_code ec;
+      const auto savegamePath = engine.getSavegamePath(newestLevelTs.slot);
+      BOOST_LOG_TRIVIAL(info) << "Deleting slot " << newestLevelTs.slot;
```
Comment (serishema) [inline src/menu/savegamecleanup.cpp:172]:
```suggestion
      const auto savegamePath = engine.getSavegamePath(slot);
      BOOST_LOG_TRIVIAL(info) << "Deleting slot " << slot;
```

File: src/ui/widgets/messagebox.h [hunk 2]
```diff
@@ -38,6 +40,11 @@ class MessageBox : public Widget
     return m_confirmed;
   }
```
Comment (serishema) [inline src/ui/widgets/messagebox.h:41]:
```suggestion
}
```

Comment (stohrendorf) [reply src/ui/widgets/messagebox.h:41]: Are you sure about that? Looks legit to me.

File: src/menu/savegamecleanup.cpp (hunk 1 above)
Comment (stohrendorf) [reply src/menu/savegamecleanup.cpp]: Brain fart fixed.

**Review by serishema — APPROVED** [review]
All good now

**Pull request closed without merging**